### PR TITLE
tests: add test for zero config

### PR DESCRIPTION
**Pull request by anshumanv**
**What kind of change does this PR introduce?**

tests

**Did you add tests for your changes?**
Yes

**If relevant, did you update the documentation?**
NA

**Summary**
- Add some test cases for zero-config.

**Does this PR introduce a breaking change?**
No

**Other information**
- Any suggestion on how to strengthen are appreciated.

### CODE REVIEW

File: test/zero-config/with-config/zero-config-entry.test.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+const { run } = require('../../utils/test-utils');
+
+describe('Zero Config', () => {
+    it('runs when config is present but not supplied via flag', () => {
+        const { stdout, stderr } = run(__dirname, [], false);
+        console.log({ stdout, stderr });
```
Comment (jamesgeorge007) [inline test/zero-config/with-config/zero-config-entry.test.js:6]: Unnecessary log.

File: test/zero-config/entry-present/zero-config.test.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+const { run } = require('../../utils/test-utils');
+
+describe('Zero Config tests', () => {
+    it('runs when no config is supplied but entry is present', () => {
+        const { stdout, stderr } = run(__dirname, [], false);
+        // Should be able to find the entry file
+        expect(stdout).not.toContain("Module not found: Error: Can't resolve './src'");
```
Comment (evilebottnawi) [inline test/zero-config/entry-present/zero-config.test.js:7]: No need test it

File: test/zero-config/with-config/zero-config-entry.test.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+const { run } = require('../../utils/test-utils');
+
+describe('Zero Config', () => {
+    it('runs when config is present but not supplied via flag', () => {
+        const { stdout, stderr } = run(__dirname, [], false);
+        // Should find the entry file, thus not throw the below warning
+        expect(stdout).not.toContain("Module not found: Error: Can't resolve './src'");
```
Comment (evilebottnawi) [inline test/zero-config/with-config/zero-config-entry.test.js:7]: No need test it

File: test/zero-config/entry-present/zero-config.test.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+const { run } = require('../../utils/test-utils');
+
+describe('Zero Config tests', () => {
+    it('runs when no config is supplied but entry is present', () => {
+        const { stdout, stderr } = run(__dirname, [], false);
+        // Should be able to find the entry file
+        expect(stdout).toContain('./src/index.js');
+        // Should output at the default output dir and filename
+        expect(stdout).toContain('Entrypoint main = main.js');
```
Comment (anshumanv) [inline test/zero-config/entry-present/zero-config.test.js:9]: We already have it, only entry present and config absent. @evilebottnawi

**Comment by evilebottnawi** [comment]
Can you provide here structure what we generate on init by default?

**Comment by anshumanv** [comment]
on init? as in?

**Comment by anshumanv** [comment]
The default config you mean?

**Comment by evilebottnawi** [comment]
@anshumanv `init`, we need investigate unnecessary stuff

**Comment by anshumanv** [comment]
> @anshumanv init, we need investigate unnecessary stuff

init tests are here - https://github.com/webpack/webpack-cli/pull/1419

**Comment by evilebottnawi** [comment]
@anshumanv Feel free to merge after CI green

**Comment by anshumanv** [comment]
Cool, need another review then I'll merge once green. 👍

File: test/zero-config/entry-present/zero-config.test.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
+const { run } = require('../../utils/test-utils');
+
+describe('Zero Config tests', () => {
+    it('runs when no config is supplied but entry is present', () => {
```
Comment (rishabh3112) [inline test/zero-config/entry-present/zero-config.test.js:4]: We could also test for existence of bundle output file.

Comment (anshumanv) [reply test/zero-config/entry-present/zero-config.test.js:4]: We can def do that, thinking if this would be the right scope for it as output file check is something which is config/flag agnostic. wdyt? Since we're not checking for it anywhere else, should be also checked for it in all other tests?

Comment (anshumanv) [reply test/zero-config/entry-present/zero-config.test.js:4]: Or maybe we can just check the test util since it's used everywhere, will add tests for the `run` fn that we use in all tests separately. Would make more sense to check file existence there.

Comment (rishabh3112) [reply test/zero-config/entry-present/zero-config.test.js:4]: In this particular case we should test it either way imo.  Because the test is for checking whether without config we are bundling or not.

Tests for run as well in a different PR would be great :)

Comment (jamesgeorge007) [reply test/zero-config/entry-present/zero-config.test.js:4]: > Or maybe we can just check the test util since it's used everywhere, will add tests for the `run` fn that we use in all tests separately. Would make more sense to check file existence there.

We're consuming the `run` fn for couple of test cases where bundle output is not generated, say the `migrate` command. It should be taken into account while writing the respective tests.

Comment (anshumanv) [reply test/zero-config/entry-present/zero-config.test.js:4]: Sounds fair, added. 👍
